### PR TITLE
Backend [Integration Test] bounty.go handlers GetPersonCreatedBounties, GetPersonAssignedBounties, & GetBountyByCreated

### PR DESCRIPTION
**Pull request by AbdulWahab3181**
…tBountyByCreated

## Describe your changes

Added Unit test for bounty handler to test GetPersonCreatedBounties, GetPersonAssignedBounties and GetBountyByCreated methods

## Issue ticket number and link

Closes https://github.com/stakwork/sphinx-tribes/issues/1509

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have tested on Chrome and Firefox
- [ ] I have tested on a mobile device
- [x] I have provided a screenshot or recording of changes in my PR if there were updates to the frontend

### CODE REVIEW

File: handlers/bounty_test.go [hunk 1]
```diff
@@ -485,6 +507,15 @@ func TestGetBountyByCreated(t *testing.T) {
...
 		assert.Equal(t, http.StatusOK, rr.Code)
+
+		var responseData []db.BountyResponse
+		err = json.Unmarshal(rr.Body.Bytes(), &responseData)
+		if err != nil {
+			t.Fatalf("Error decoding JSON response: %s", err)
+		}
+
+		assert.NotEmpty(t, responseData)
+		assert.Len(t, responseData, 1)
```
Comment (kevkevinpal) [inline handlers/bounty_test.go:518]: noticed that we dont assert the exact value here

File: handlers/bounty_test.go [hunk 2]
```diff
@@ -563,3 +594,183 @@ func TestGetOrganizationPreviousBountyByCreated(t *testing.T) {
...
+		assert.Equal(t, http.StatusOK, rr.Code)
+
+		var responseData []db.BountyResponse
+		err = json.Unmarshal(rr.Body.Bytes(), &responseData)
+		if err != nil {
+			t.Fatalf("Error decoding JSON response: %s", err)
+		}
+
+		assert.Empty(t, responseData)
+		assert.Len(t, responseData, 0)
```
Comment (kevkevinpal) [inline handlers/bounty_test.go:774]: same with this assertion

**Review by kevkevinpal — COMMENTED** [review]
I think for the most part this PR is good, would be better if we were asserting the exact response but we're already mocking the db so not sure how good of a test that would be.

I need to double check that we're testing the failure scenarios since those are important aswell

File: handlers/bounty_test.go (hunk 1 above)
Comment (AbdulWahab3181) [reply handlers/bounty_test.go:518]: @kevkevinpal [Added](https://github.com/stakwork/sphinx-tribes/pull/1544/commits/9305d7298863a25d8377daa60e4e3d50031b0496#diff-f47eb74033ab155c9b6163240180949abb75b2fd4a1dcd61d0a49d0278500aeaR519-R525) additional assertions to validate the response

File: handlers/bounty_test.go [hunk 3]
```diff
@@ -563,3 +594,183 @@ func TestGetOrganizationPreviousBountyByCreated(t *testing.T) {
...
+		assert.Equal(t, http.StatusOK, rr.Code)
+
+		var responseData []db.BountyResponse
+		err = json.Unmarshal(rr.Body.Bytes(), &responseData)
+		if err != nil {
+			t.Fatalf("Error decoding JSON response: %s", err)
+		}
+
+		assert.NotEmpty(t, responseData)
+		assert.Len(t, responseData, 2)
```
[the remark this answers is not in the document]
Comment (AbdulWahab3181) [reply handlers/bounty_test.go:664]: @kevkevinpal [Added](https://github.com/stakwork/sphinx-tribes/pull/1544/commits/9305d7298863a25d8377daa60e4e3d50031b0496#diff-f47eb74033ab155c9b6163240180949abb75b2fd4a1dcd61d0a49d0278500aeaR666-R669) additional assertions to validate the exact response.

File: handlers/bounty_test.go [hunk 4]
```diff
@@ -563,3 +594,183 @@ func TestGetOrganizationPreviousBountyByCreated(t *testing.T) {
...
```
[the remark this answers is not in the document]
Comment (AbdulWahab3181) [reply handlers/bounty_test.go:761]: @kevkevinpal [Added](https://github.com/stakwork/sphinx-tribes/pull/1544/commits/9305d7298863a25d8377daa60e4e3d50031b0496#diff-f47eb74033ab155c9b6163240180949abb75b2fd4a1dcd61d0a49d0278500aeaR762-R766) additional assertions to validate the response

File: handlers/bounty_test.go (hunk 2 above)
Comment (AbdulWahab3181) [reply handlers/bounty_test.go:795]: I think there is not need as the response data is already empty

**Comment by ecurrencyhodler** [comment]
Paid 200k sats! https://community.sphinx.chat/bounty/1479

**Pull request merged**